api/v1beta1/nutanix_types.go

```diff
@@ -61,6 +61,8 @@ const (
 
 // NutanixResourceIdentifier holds the identity of a Nutanix PC resource (cluster, image, subnet, etc.)
 // +union
+// +kubebuilder:validation:XValidation:rule="self.type == 'name' ? (has(self.name) && size(self.name) > 0) : !has(self.name)",message="'name' must be set and non-empty when type is 'name', and forbidden otherwise"
```

Comment:
Min length of 1 for name can be added via kubebuilder validation annotation on `Name` property.

UUID pattern can be added via kubebuilder validation annotation on UUID property.

I would also add multiple CEL rules with more specific messages that just deals with the semantics

```suggestion
// +kubebuilder:validation:XValidation:rule="self.type != 'name' || has(self.name)",message="the 'name' field must be present when the 'type' is 'name'"
// +kubebuilder:validation:XValidation:rule="self.type != 'uuid' || has(self.uuid)",message="'the 'uuid' field must be present when the 'type' is 'uuid'"
// +kubebuilder:validation:XValidation:rule="self.type != 'name' || !has(self.uuid)",message="the 'uuid' field cannot be set when the 'type' is 'name'"
// +kubebuilder:validation:XValidation:rule="self.type == 'uuid' && !has(self.name)",message="the 'name' field cannot be set when the 'type' is 'uuid'"
```